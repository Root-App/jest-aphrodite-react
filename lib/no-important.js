'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
var replaceClassNames_1 = require('./replaceClassNames');
exports.replaceClassNames = replaceClassNames_1.replaceClassNames;
var serializer_1 = require('./serializer');
exports.createSerializer = serializer_1.createSerializer;
var aphroditeSerializer = serializer_1.serializer(false);
exports.aphroditeSerializer = aphroditeSerializer;
var test = aphroditeSerializer.test,
  print = aphroditeSerializer.print;
exports.test = test;
exports.print = print;
