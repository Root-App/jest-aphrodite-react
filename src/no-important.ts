import { ClassNameReplacer, replaceClassNames } from './replaceClassNames';
import { createSerializer, serializer, SerializerOptions } from './serializer';

const aphroditeSerializer = serializer(false);
const { test, print } = aphroditeSerializer;

export {
  aphroditeSerializer,
  createSerializer,
  replaceClassNames,
  print,
  test,
  ClassNameReplacer,
  SerializerOptions,
};
