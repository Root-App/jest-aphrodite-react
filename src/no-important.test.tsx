import { css, StyleSheet } from 'aphrodite/no-important';
import * as React from 'react';
import * as reactTestRenderer from 'react-test-renderer';
import { aphroditeSerializer } from './no-important';

expect.addSnapshotSerializer(aphroditeSerializer);

const Wrapper: React.SFC = props => {
  const styles = StyleSheet.create({
    wrapper: {
      background: 'papayawhip',
      padding: '4em',
    },
  });
  return <section className={css(styles.wrapper)} {...props} />;
};

const Title: React.SFC = props => {
  const styles = StyleSheet.create({
    title: {
      color: 'palevioletred',
      fontSize: '1.5em',
      textAlign: 'center',
    },
  });
  return <h1 className={css(styles.title)} {...props} />;
};

test('react-test-renderer', () => {
  const tree = reactTestRenderer
    .create(
      <Wrapper>
        <Title>
          Hello World, this is my first component styled with aphrodite!
        </Title>
      </Wrapper>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('no-important exports match default package exports', () => {
  const defaultExports = require('.');
  const noImportantExports = require('./no-important');
  expect(Object.keys(defaultExports)).toEqual(Object.keys(noImportantExports));
});
