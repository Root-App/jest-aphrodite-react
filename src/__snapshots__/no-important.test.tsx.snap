// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-test-renderer 1`] = `
.wrapper {
  background: papayawhip;
  padding: 4em;
}

.title {
  color: palevioletred;
  font-size: 1.5em;
  text-align: center;
}

<section
  className="wrapper"
>
  <h1
    className="title"
  >
    Hello World, this is my first component styled with aphrodite!
  </h1>
</section>
`;
